test/Datadog.Trace.Tests/Propagators/B3SpanContextPropagatorTests.cs

```diff
@@ -0,0 +1,201 @@
+        {
+            _propagator = new B3SpanContextPropagator();
+        }
+
+        [Theory]
+        [MemberData(nameof(GetHeaderCollectionImplementations))]
+        internal void HttpRequestMessage_InjectExtract_Identity(IHeadersCollection headers)
+        {
+            const ulong traceId = 18446744073709551615;
+            const ulong spanId = 18446744073709551614;
+            const SamplingPriority samplingPriority = SamplingPriority.AutoKeep;
+
+            var context = new SpanContext(traceId, spanId, samplingPriority);
+
+            _propagator.Inject(context, headers);
+
+            AssertExpected(headers, B3HttpHeaderNames.B3TraceId, "ffffffffffffffff");
+            AssertExpected(headers, B3HttpHeaderNames.B3SpanId, "fffffffffffffffe");
+            AssertExpected(headers, B3HttpHeaderNames.B3Sampled, "1");
+            AssertMissing(headers, B3HttpHeaderNames.B3ParentId);
+            AssertMissing(headers, B3HttpHeaderNames.B3Flags);
+
+            var resultContext = _propagator.Extract(headers);
+
+            Assert.NotNull(resultContext);
+            Assert.Equal(context.SpanId, resultContext.SpanId);
+            Assert.Equal(context.TraceId, resultContext.TraceId);
+            Assert.Equal(context.SamplingPriority, resultContext.SamplingPriority);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetHeaderCollectionImplementations))]
+        internal void HttpRequestMessage_InjectExtract_Identity_WithParent(IHeadersCollection headers)
+        {
+            const ulong traceId = 18446744073709551615;
+            const ulong spanId = 18446744073709551614;
+            const SamplingPriority samplingPriority = SamplingPriority.UserKeep;
+
+            var parentContext = new SpanContext(traceId, spanId, samplingPriority);
+            var traceContext = new TraceContext(null)
+            {
+                SamplingPriority = samplingPriority
+            };
+
+            var context = new SpanContext(parentContext, traceContext, null);
+
+            _propagator.Inject(context, headers);
+
+            AssertExpected(headers, B3HttpHeaderNames.B3TraceId, "ffffffffffffffff");
+            AssertExpected(headers, B3HttpHeaderNames.B3SpanId, context.SpanId.ToString("x16"));
+            AssertExpected(headers, B3HttpHeaderNames.B3ParentId, "fffffffffffffffe");
+            AssertExpected(headers, B3HttpHeaderNames.B3Flags, "1");
+            AssertMissing(headers, B3HttpHeaderNames.B3Sampled);
+
+            var resultContext = _propagator.Extract(headers);
+
+            Assert.NotNull(resultContext);
+            Assert.Equal(context.SpanId, resultContext.SpanId);
+            Assert.Equal(context.TraceId, resultContext.TraceId);
+            Assert.Equal(samplingPriority, resultContext.SamplingPriority);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetHeaderCollectionImplementations))]
+        internal void WebRequest_InjectExtract_Identity(IHeadersCollection headers)
```

Comment:
This looks like a duplicate of `HttpRequestMessage_InjectExtract_Identity` minus the changes in the traceId, spanId, and samplingPriority. Perhaps this method can be removed or just renamed so it more clearly expresses the intent of the test

Reply:
Great catch!